api/repositories/shared.go

```diff
@@ -1,15 +1,56 @@
 package repositories
 
 import (
+	"context"
 	"errors"
+	"fmt"
 
 	"code.cloudfoundry.org/cf-k8s-controllers/api/authorization"
 	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var CFAppGVR = schema.GroupVersionResource{
+	Group:    "workloads.cloudfoundry.org",
+	Version:  "v1alpha1",
+	Resource: "cfapps",
+}
+
+type NamespaceRetriever struct {
```

Comment:
I would put this object along with its method in a separate file. Putting all sorts of utilites in a shared file would result in the file becoming huge and hard to orient in.

Reply:
+1 for grouping similar objects into separate file. `shared.go` sound like `oh, I don't know where to put that, let me shove it in shared`